project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module

```diff
@@ -685,3 +691,87 @@ function c4m_og_get_user_group_nodes_by_bundle(
   }
   return $groups;
 }
+
+/**
+ * Implements hook_node_update().
+ */
+function c4m_og_node_update($node) {
+  if ($node->type != 'group') {
+    return;
+  }
+
+  c4m_og_open_group_access_activate_pending_members($node);
+}
+
+/**
+ * Activate pending members when access level of a group is changed.
+ *
+ * When changing the access level from 'Moderated' to 'Open', all pending
+ * members should become active group members.
+ *
+ * @param object $node
+ *   The node object while being updated.
+ *
+ * @throws \Exception
+ *   When the node object doesn't contain the 'original' property.
+ */
+function c4m_og_open_group_access_activate_pending_members($node) {
+  if (empty($node->original)) {
+    throw new Exception('Node object must contain the original property.');
```

Comment:
Basically this should never happens, but in case that in the future someone will use this function not via a _hook_node_update_ and/or the `original` property is missing then we must throw this exception.
See this as a precaution.